### PR TITLE
Added classical Memory Game sample

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
Here is another one, I am really bad at this :smile:

memory-game gif

### CODE REVIEW

**Comment by forki** [comment]
keep them coning!

File: samples/react/MemoryGame/src/State.fs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+let cardSelected id (cards: Card list) = 
+    let card = List.find (fun c -> c.Id = id) cards
+    card.Selected
+
+let gameCleared (model: Model) =  
+    List.forall (fun card -> card.MatchFound) model.Cards
+
+let rec update action model  =
```
Comment (forki) [inline samples/react/MemoryGame/src/State.fs]: urgs why is the update function recursive?

File: samples/react/MemoryGame/src/State.fs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+open MemoryGame.Types
+open System
+let random = new Random();
+
+let getCards() = 
+    let images = [ "violin"; "electric-guitar"; "headphones"; "piano"; "saxophone";  "trumpet";"turntable";"bass-guitar" ]
+    images 
+    |> List.append images
+    |> List.sortBy (fun img -> random.Next())
```
Comment (forki) [inline samples/react/MemoryGame/src/State.fs]: love it ;-)

File: samples/react/MemoryGame/src/State.fs (hunk 1 above)
Comment (Zaid-Ajaj) [inline samples/react/MemoryGame/src/State.fs]: lol I have no idea :smile: , I threw it away

**Comment by et1975** [comment]
Thanks!

**Pull request merged**